package.json

```diff
@@ -1,13 +1,82 @@
 {
-  "name": "vibc-core-smart-contracts",
-  "version": "1.0.0",
-  "main": "index.js",
-  "repository": "https://github.com/open-ibc/vibc-core-smart-contracts",
+  "name": "@open-ibc/vibc-core-smart-contracts",
+  "version": "0.1.3",
+  "main": "dist/index.js",
+  "bin": {
+    "deploy-vibc-core-smart-contracts": "./dist/scripts/deploy-script.js",
+    "upgrade-vibc-core-smart-contracts": "./dist/scripts/upgrade-script.js"
+  },
   "license": "MIT",
   "dependencies": {
-    "solhint": "^4.1.1"
+    "@commander-js/extra-typings": "^12.0.1",
+    "@typechain/ethers-v6": "^0.5.0",
+    "ethers": "^6.7.1",
+    "nunjucks": "^3.2.4",
+    "solhint": "^4.1.1",
+    "typechain": "^8.3.2",
+    "winston": "^3.13.0",
+    "yaml": "^2.4.1",
+    "zod": "^3.23.4",
+    "zx": "^8.0.2"
+  },
+  "devDependencies": {
+    "@nomicfoundation/hardhat-chai-matchers": "^2.0.0",
+    "@nomicfoundation/hardhat-ethers": "^3.0.0",
+    "@nomicfoundation/hardhat-ignition": "^0.15.0",
+    "@nomicfoundation/hardhat-ignition-ethers": "^0.15.0",
+    "@nomicfoundation/hardhat-network-helpers": "^1.0.0",
+    "@nomicfoundation/hardhat-toolbox": "^5.0.0",
+    "@nomicfoundation/hardhat-verify": "^2.0.0",
+    "@typechain/hardhat": "^9.0.0",
+    "@types/nunjucks": "^3.2.6",
+    "@types/winston": "^2.4.4",
+    "chai": "^4.2.0",
+    "hardhat": "^2.14.0",
+    "hardhat-gas-reporter": "^1.0.8",
+    "solidity-coverage": "^0.8.0",
+    "tsup": "^8.0.2"
   },
   "scripts": {
-    "lint": "solhint contracts/**/*.sol"
+    "lint": "solhint contracts/**/*.sol",
+    "test": "forge test",
+    "build": "npm run gen-types && tsup",
+    "gen-types": "npm run build-contracts && typechain --target ethers-v6 --out-dir src/evm/contracts/ './out/?(OptimisticProofVerifier|ProofVerifier|Ibc|IbcUtils|Channel|Dispatcher|Mars|Earth|UniversalChannelHandler|DummyProofVerifier|DummyLightClient|ERC1967Proxy|OptimisticLightClient).sol/*.json'",
```

Comment:
can this be `generate-ts-bindings` so we can then have another one saying `generate-golang-bindings`?

Reply:
Yes I can rename it; though `generate-golang-bindings` would be run by `abigen` right? So Since we have to run `gen-types` with npm, I actually don't think we would have a naming collision?

Reply:
It shouldn't matter what command you use, right? these names are the "npm script names". You should end up with something like this 
```json
    "generate-ts-bindings": "npm run build-contracts && typechain --target ethers-v6 --out-di...",
    "generate-golang-bindings": "npm run build-contracts && abigen --abi ...",
```

Reply:
ah gotcha - I was initially confused since I wasn't sure why we would need an npm command to generate go bindings; but this seems to be a general practice. So for now I can rename just in case. 

Renamed gen-types to build-ts-contract-bindings and added a dummy build-go-contract-bindings for now! Thanks for the feedback!